test/metabase/lib/stage_test.cljc

```diff
@@ -36,7 +41,7 @@
                  :database     (meta/id)
                  :lib/metadata meta/metadata-provider}]
       (is (mr/validate ::lib.schema/query query))
-      (is (=? [(merge (meta/field-metadata :venues :price)
+      (is (=? [(merge (m/filter-vals some? (meta/field-metadata :venues :price))
```

Comment:
This seems to be appearing in lots of places - if we're now removing nils in some part of metadata processing, shouldn't we fix `meta/field-metadata` to return what we would see from `visible-columns` on a basic table query?

Reply:
`meta/field-metadata` is supposed to mimic what you'd get from `metabase.lib.metadata/field` with the app DB metadata provider, I didn't want to change that. The fact that `nil` keys are getting removed is mostly a side effect of the new field resolution code and it didn't seem super important to change that in a way that didn't require updating a handful of tests.